Move case studies routes

georges1996 opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

This PR https://github.com/alphagov/frontend/pull/4379 needs to be merged in and this will need to be rebased to fix the case studies which have translations e.g https://www.gov.uk/government/case-studies/doing-business-in-spain
## What
Handle Case Studies documents in Frontend, in preparation for them being taken out of Government Frontend


## Why

Ticket: https://trello.com/c/ybpXwp34/368-move-document-type-casestudy-from-government-frontend-to-frontend, [Jira issue PNP-9253](https://gov-uk.atlassian.net/browse/PNP-9253)

## How

By moving the routes and necessary code to get the pages rendering in the frontend application 

## Screenshots?
# Frontend 
<img width="1728" alt="Screenshot 2024-11-06 at 08 22 40" src="https://github.com/user-attachments/assets/7bcb6d41-21d9-41e2-8924-2fefef3168ab">
<img width="1728" alt="Screenshot 2024-11-06 at 08 23 02" src="https://github.com/user-attachments/assets/d35fdab1-d2ab-481a-b55e-0a75aa5acdb5">

# Gov-Frontend
<img width="1726" alt="Screenshot 2024-11-06 at 08 22 34" src="https://github.com/user-attachments/assets/41ed8105-0881-4f2d-af00-26cbd31195e9">
<img width="1728" alt="Screenshot 2024-11-06 at 08 22 54" src="https://github.com/user-attachments/assets/d1e53358-22fa-4922-a83e-70a25c7fb19d">


Example case studies check: 
- https://govuk-frontend-app-pr-4372.herokuapp.com/government/case-studies/doing-business-in-spain
- https://govuk-frontend-app-pr-4372.herokuapp.com/government/case-studies/terence-age-33-stoke-on-trent
- https://govuk-frontend-app-pr-4372.herokuapp.comgovernment/case-studies/vdec-is-supporting-a-gbs-vaccine-to-prevent-newborn-deaths
- https://govuk-frontend-app-pr-4372.herokuapp.com/government/case-studies/2013-elections-in-swaziland

georges1996 committed:
Update README with a sample of case studies routes

georges1996 committed:
Add case-studies example

georges1996 committed:
Add Updatable concern and tests

Commit audit trail:
- https://github.com/alphagov/government-frontend/blob/a938e4bee78b794e4e5303a1bde1dfd05e94b287/app/presenters/content_item/updatable.rb
- https://github.com/alphagov/government-frontend/blob/a938e4bee78b794e4e5303a1bde1dfd05e94b287/test/presenters/content_item/updatable_test.rb

georges1996 committed:
Update Withdrawable concern and tests

Commit audit trail:
- https://github.com/alphagov/government-frontend/blob/a938e4bee78b794e4e5303a1bde1dfd05e94b287/app/presenters/content_item/withdrawable.rb
- https://github.com/alphagov/government-frontend/blob/a938e4bee78b794e4e5303a1bde1dfd05e94b287/test/presenters/content_item/withdrawable_test.rb

georges1996 committed:
Add missing test for the native_language_name_for method

georges1996 committed:
Update content_item model

- To expose first_public_at and first_published_at attributes
- Add Organisations concern
- Add contributors method which calls the organisations function from Organisations concern
to allow us to have a default value for all document_types

We need both first_public_at and first_published_at as these both need
to be checked otherwise updates won't be picked up for the other document types
that use this concern.

We have added a contributors function which returns the list of contributors,
these are then rendered in the "from" section of the publisher_metadata partial.
The contributors function will default to return just organisations for all document_types
however this function will be overriden depending on what each document_type needs.

georges1996 committed:
Add publisher_metadata partial and scss

Commit audit trail:
- https://github.com/alphagov/government-frontend/blob/0aee348d51cebacd5344115f2ca02bdb17095249/app/views/shared/_publisher_metadata_with_logo.html.erb
- https://github.com/alphagov/government-frontend/blob/0aee348d51cebacd5344115f2ca02bdb17095249/app/assets/stylesheets/helpers/_publisher-metadata-with-logo.scss

As we dont need the logo part of the partial for case studies, I have removed it and renamed the partial

georges1996 committed:
Add case study specific files

Add case studies model, view, controller, route and tests as well as content_item_model_presenter and related concerns

Commit audit trail:
- https://github.com/alphagov/government-frontend/blob/0aee348d51cebacd5344115f2ca02bdb17095249/app/presenters/case_study_presenter.rb
- https://github.com/alphagov/government-frontend/blob/0aee348d51cebacd5344115f2ca02bdb17095249/app/views/content_items/case_study.html.erb
- https://github.com/alphagov/government-frontend/blob/0aee348d51cebacd5344115f2ca02bdb17095249/test/integration/case_study_test.rb
- https://github.com/alphagov/government-frontend/blob/0aee348d51cebacd5344115f2ca02bdb17095249/test/presenters/case_study_presenter_test.rb
- https://github.com/alphagov/government-frontend/blob/0aee348d51cebacd5344115f2ca02bdb17095249/config/routes.rb

Update app/views/components/_published_dates.html.erb to render dates correctly

georges1996 committed:
Add case_study locale

Using govuk-docker-run rake "consolidation:copy_translation[content_item.schema_name.case_study]"
we port over the case_study translations and change it to be under our new yaml structure

Audit trail:
- https://github.com/alphagov/government-frontend/tree/0aee348d51cebacd5344115f2ca02bdb17095249/config/locales

KludgeKML commented on spec/models/concerns/linkable_spec.rb:
```diff
@@ -0,0 +1,51 @@
+  end
+end
+
+RSpec.describe Linkable, type: :model do
```
You shouldn't need `type: model` here - it's inferred from the file location.

KludgeKML commented on app/models/concerns/withdrawable.rb:
```diff
@@ -0,0 +1,45 @@
+    content_store_hash["withdrawn_notice"]
+  end
+
+  def withdrawal_notice_title
+    "This #{withdrawal_notice_context.downcase} was withdrawn on #{withdrawal_notice_time}".html_safe
+  end
+
+  def withdrawal_notice_context
+    I18n.t("formats.#{schema_name}", count: 1, locale: :en)
+  end
+
+  def withdrawal_notice_time
+    ActionController::Base.helpers.tag.time(
+      english_display_date(withdrawal_notice["withdrawn_at"]),
+      datetime: withdrawal_notice["withdrawn_at"],
+    )
+  end
+
+  def english_display_date(timestamp, format = "%-d %B %Y")
+    I18n.l(Time.zone.parse(timestamp), format:, locale: :en) if timestamp
+  end
```
A bit suspicious of these presentational items being in a model concern, might be better as helpers?

leenagupte replied:
Travel Advice and Guides need to know the withdrawn state too, but I've tackled it a bit differently by splitting out the model and presentational concerns:

model concern: https://github.com/alphagov/frontend/pull/4225/commits/0dfd81189601511ab3d468001b35098ffec32cad

generic presenter: https://github.com/alphagov/frontend/pull/4225/commits/0590ed72bf4543ff55feae6b1539f817fb54b3b5

travel advice presenter: https://github.com/alphagov/frontend/pull/4225/commits/53ee7c310524d77dc7c83fe86ed10d5d23b2cf8d

I'm not completely happy with the approach I've taken either, so I think it would be good to have a discussion about this.

KludgeKML commented on app/helpers/locale_helper.rb:
```diff
@@ -7,6 +7,16 @@ def page_text_direction
+      I18n.default_locale
+    end
+  end
+
```
This is just... moving the method? Is it necessary?

KludgeKML commented on app/models/content_item.rb:
```diff
@@ -1,9 +1,11 @@
 require "ostruct"
 
 class ContentItem
+  include Withdrawable
```
Is Withdrawable definitely something that all ContentItems (or, let's say, the overwhelming majority of them) are going to use?

KludgeKML commented on app/controllers/case_studies_controller.rb:
```diff
@@ -0,0 +1,11 @@
+    @case_study = CaseStudiesPresenter.new(@content_item, view_context)
+  end
+
+private
+
+  def content_item_slug
+    request.path
+  end
+end
```
I think we can remove this now (all ContentItems use request.path)

KludgeKML commented on app/presenters/case_studies_presenter.rb:
```diff
@@ -0,0 +1,18 @@
+    @view_context = view_context
+  end
+
+  def title_and_context
+    {
+      title: content_item.title,
+      context: I18n.t("formats.#{content_item.document_type}", count: 1),
+      context_locale: view_context.t_locale_fallback("formats.#{content_item.document_type}", count: 1),
+      average_title_length: "long",
+    }
+  end
+end
```
It might be worth just putting this straight into the view (take a look at the take_part controller), which would allow you to cut out the entire presenter.

KludgeKML commented on app/views/case_studies/show.html.erb:
```diff
@@ -0,0 +1,52 @@
+  <%= render "govuk_publishing_components/components/machine_readable_metadata", { content_item: @content_item.to_h, schema: :article } %>
+<% end %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds responsive-top-margin">
+    <%= render 'govuk_publishing_components/components/title', @case_study.title_and_context %>
+  </div>
+  <%= render 'shared/translations' %>
+  <div class="govuk-grid-column-two-thirds">
+    <%= render 'govuk_publishing_components/components/lead_paragraph', text: @content_item.description %>
+  </div>
+</div>
+
```
You might need to look at the take part PR around adding a main class here (this might also affect help pages, FYI)

KludgeKML commented on spec/helpers/locale_helper_spec.rb:
```diff
@@ -33,7 +33,7 @@
     it "returns default locale for a string with no locale translation" do
       I18n.with_locale(:cy) do
-        expect(t_locale_fallback("formats.big_decision", count: 1, locale: :cy)).to eq(:en)
+        expect(t_locale_fallback("content_item.schema_name.case_study.no_locale", count: 1, locale: :cy)).to eq(:en)
```
Is this right? I think we want the case_study locales to go into `formats.case_study.<whatever>`

KludgeKML commented on app/models/content_item.rb:
```diff
@@ -20,6 +22,8 @@ def initialize(content_store_response, override_content_store_hash: nil)
     @locale = content_store_hash["locale"]
     @public_updated_at = content_store_hash["public_updated_at"]
+    @first_published_at = content_store_hash["first_published_at"]
+    @first_public_at = content_store_hash.dig("details", "first_public_at") || first_published_at
```
Do we know how many different schemas have a `first_public_at`? (is it only case studies?)

KludgeKML commented on app/presenters/case_study_presenter.rb:
I don't know if we should make empty presenters (if it needs a presenter but it's just the standard ContentItemModelPresenter, let's use that in the controller).

KludgeKML commented on app/models/concerns/metadata.rb:
```diff
@@ -0,0 +1,29 @@
+        last_updated: updated,
+      }
+
+      unless pending_stats_announcement?
```
Can we add a test that covers this? (I think we're not testing when pending_stats_announcements exist)

KludgeKML commented on app/models/concerns/updatable.rb:
```diff
@@ -0,0 +1,40 @@
+    if public_updated_at && first_public_at
+      Time.zone.parse(public_updated_at) != Time.zone.parse(first_public_at)
+    else
+      false
```
We're not testing this branch, could we add a test for it?

KludgeKML commented on app/presenters/withdrawable_presenter.rb:
```diff
@@ -0,0 +1,37 @@
+  def withdrawal_notice_component
+    {
+      title: withdrawal_notice_title,
+      description_govspeak: content_item.withdrawal_notice["explanation"]&.html_safe,
```
We're not testing if the explanation key doesn't exist.

KludgeKML commented on app/presenters/withdrawable_presenter.rb:
```diff
@@ -0,0 +1,37 @@
+      title: withdrawal_notice_title,
+      description_govspeak: content_item.withdrawal_notice["explanation"]&.html_safe,
+      time: withdrawal_notice_time,
+      lang: I18n.locale.to_s == "en" ? false : "en",
```
We're not testing the "en" branch (I think? we're definitely not testing one of the branches here)

leenagupte reviewed (changes requested):
I think this has made some good progress.

I have a few concerns about the split between presentational and model code, and think some of the model concerns need to be looked again. I think it would get good to talk about this in person.

deborahchua commented on app/views/case_studies/show.html.erb:
```diff
@@ -0,0 +1,59 @@
+    </div>
+  </div>
+
+  <hr class="govuk-section-break govuk-section-break--s govuk-section-break--visible" aria-hidden="true">
```
The border is defined in https://github.com/alphagov/government-frontend/blob/main/app/assets/stylesheets/helpers/_publisher-metadata-with-logo.scss so I think once it's added in https://github.com/alphagov/frontend/pull/4372/commits/26a483dd6224a6d6edbda1925b80c8eeb07d16c1 this can be removed.

KludgeKML commented on spec/support/concerns/updatable.rb:
```diff
@@ -0,0 +1,51 @@
+    expect(described_class.new(@example_doc).updated).to eq "2024-07-16T12:36:44Z"
+  end
+
+  it "knows it's history" do
```
type: it's -> its

KludgeKML commented on app/assets/stylesheets/components/_publisher-metadata-with-logo.scss:
```diff
@@ -0,0 +1,24 @@
+@import "govuk_publishing_components/individual_component_support";
```
I think we might have to find somewhere different for this file - if something's a component, it has to meet a set of criteria (it has to have a specific set of component tests, docs, the class names for stylesheet matching have to have an app-c- prefix, etc). This wasn't a component in government-frontend, so perhaps we can find a better place for it in frontend. (maybe we can just have a stylesheets/views/shared folder?)

leenagupte commented on app/views/components/_published_dates.html.erb:
```diff
@@ -34,13 +34,14 @@
         </ol>
       </div>
     <% end %>
+    <hr class="govuk-section-break govuk-section-break--l govuk-section-break">
```
Why did this HR need to be added to the component?

leenagupte reviewed (changes requested):
I've added some inline comments.

My biggest area for concern is still the Metadata concern and how it's used. It would be good to talk about it in person and come up with a plan.

georges1996 committed:
Add govuk_styled_links_list helper method

- https://github.com/alphagov/government-frontend/blob/2e602630db112fe0594653cebbeb313396e67a02/app/presenters/content_item/linkable.rb

As part of this commit I needed to change the govuk_styled_link method as
it was causing tests to fail in CI. This was due to some of the link titles
containing an &. This change fixes this issue and all tests pass

This is creating a helper method to allow us to move some of the presentation logic
into a helper to avoid us from either needing to recreate the presenter, or creating a model concern.

georges1996 committed:
Add Organisations and WorldwideOrganisations concerns

Due to the complexity of the Linkable presenter, we have decided to split the
Organisations and WorldwideOrganisations collections into their own concerns
this makes it easier to manage as the logic in the Linkable presenter in government-frontend
was very hard to understand and unpick.

In addition to this the Metadata presenter from government-frontend was consuming the organisations
array from the Linkable presenter, which seemed odd as a child concern should not be consuming
another child concerns methods.

This change is in preparation for allowing the ContentItem model to have a parent method to return just
organisations, we will then add an overriding method in the document type model to override this,
in this case the Case Study model will consume from both the Organisations and WorldwideOrganisations concern.

- https://github.com/alphagov/government-frontend/blob/2e602630db112fe0594653cebbeb313396e67a02/app/presenters/content_item/linkable.rb
- https://github.com/alphagov/government-frontend/blob/2e602630db112fe0594653cebbeb313396e67a02/app/presenters/content_item/metadata.rb

deborahchua commented on spec/support/concerns/organisations.rb:
```diff
@@ -0,0 +1,55 @@
+RSpec.shared_examples "it can have organisations" do |schema|
+  before do
+    @example_doc = GovukSchemas::Example.find(schema, example_name: schema)
```
I think `@example_doc` should be renamed to `@content_store_response` for consistency with the other RSpec files.

KludgeKML replied:
FYI, there's a BIG PR about changing the existing spec files - @georges1996 it would probably be useful if you temporarily added the rspec rubocop config and just run it over the files you're touching/adding in this PR.

```# .rubocop.yml
inherit_gem:
  rubocop-govuk:
    ...
    - config/rspec.yml

deborahchua commented on spec/support/concerns/organisations.rb:
```diff
@@ -0,0 +1,55 @@
+RSpec.shared_examples "it can have organisations" do |schema|
+  before do
+    @example_doc = GovukSchemas::Example.find(schema, example_name: schema)
+    @example_doc["details"]["emphasised_organisations"] = %w[11234500]
```
Do we want to hardcode an example here, when I think we want to test the data in the schema that's passed into it?

deborahchua commented on app/models/concerns/updatable.rb:
```diff
@@ -0,0 +1,44 @@
+    changes = content_store_hash.dig("details", "change_history") || []
+    changes.map do |item|
+      {
+        display_time: item["public_timestamp"],
```
This is missing the `display_date` method from https://github.com/alphagov/government-frontend/blob/a938e4bee78b794e4e5303a1bde1dfd05e94b287/app/presenters/content_item/updatable.rb#L31, or is the technical approach going forward to use it in the view? e.g. we should be including it [here](https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/app/views/components/_published_dates.html.erb#L37) instead

deborahchua commented on spec/support/concerns/updatable.rb:
```diff
@@ -0,0 +1,52 @@
+RSpec.shared_examples "it can be updated" do |schema|
```
Could we update this test so that it accepts an example name that differs to the schema name as an optional argument?

deborahchua commented on app/views/case_study/show.html.erb:
```diff
@@ -0,0 +1,71 @@
+
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <div class="content-bottom-margin">
```
I think we need to copy over the [content-bottom-margin helper stylesheet](https://github.com/alphagov/government-frontend/blob/main/app/assets/stylesheets/helpers/_content-bottom-margin.scss) and import in [application.scss](https://github.com/alphagov/government-frontend/blob/main/app/assets/stylesheets/application.scss#L17) which will add the missing spacing here (I just spotted this for call for evidence docs and also in your screenshots)

georges1996 committed:
Add withdrawn_notice.title translation

deborahchua commented on app/views/shared/_publisher_metadata.html.erb:
```diff
@@ -0,0 +1,16 @@
+  add_stylesheet_path("helpers/_publisher-metadata-with-logo.css")
+%>
+
+<div class="govuk-grid-row govuk-!-margin-top-3">
```
Do we need the class `govuk-!-margin-top-3` here? It adds some additional spacing that the page doesn't have in government-frontend

leenagupte reviewed (commented):
This is making progress. 

I've added some inline comments. They're predominantly about the information contained in the commit messages, the hard-coded elements in the shared examples and some naming concerns.

georges1996 committed:
Add content-bottom-margin scss file

Audit trail:
- Copy over stylings from https://github.com/alphagov/government-frontend/blob/90cf5386b165b3a342689744feece7c83449b153/app/assets/stylesheets/helpers/_content-bottom-margin.scss

KludgeKML commented on app/models/content_item.rb:
```diff
@@ -51,6 +54,10 @@ def available_translations
     translations.sort_by { |t| t["locale"] == I18n.default_locale.to_s ? "" : t["locale"] }
   end
 
+  def related_entities
```
(I'm not 100% sure what the actual use is here, and the code in this comment is outdated, but I'd suggest we have some kind of prinicipal where by default we don't refer to the same thing multiple times. If this method is - as implied by the link above - taking an array of organisations, perhaps it should be called **organisation_whatever** - like `organisation_links_list`, say, if it's exclusively being used to pass into `govuk_styled_links_list`)

leenagupte replied:
I think, and I hope that this is correct, that `related_entities` is the list of contributors / content creators rendered in the "from" section of the page.

I assume that "organisations" is the default list, and that's why it's been added to the the content item model. But other document types will need a slightly different list, for example a speech will need to add the speaker (person) as well as the organisation, and in that case the `related_entities` will be overridden in the child model.

leenagupte replied:
I do think that the whole object doesn't need to be passed to the helper though, it should just be a list of titles and urls.

leenagupte reviewed (commented):
Thanks for making the updates requested in the previous review.

I think there's still some work left to be done on how the Updatable concern is tested. 
I've added inline comments with detail, as well as a few other small things that should be fixed.

leenagupte commented on app/helpers/link_helper.rb:
```diff
@@ -13,6 +13,10 @@ def govuk_styled_link(text, path: nil, inverse: false)
     classes << " govuk-link--inverse" if inverse
 
-    "<a href='#{path}' class='#{classes}'>#{text}</a>".html_safe
+    "<a href='#{path}' class='#{classes}'>".html_safe + text + "</a>".html_safe
```
Was it in a separate PR?

leenagupte replied:
I'm talking about a test for "link titles containing an &"

leenagupte reviewed (commented):
Thank for making the changes requested in the previous reviews.

There are only 3 inline comments this time, all of them related to tests. The most important one is that there are no tests for the `contributors` method in the case study the model.

deborahchua commented on app/views/case_study/show.html.erb:
```diff
@@ -0,0 +1,70 @@
+<% content_for :title, "#{@case_study_presenter.page_title} - #{I18n.t("formats.#{content_item.document_type}", count: 1)} - GOV.UK" %>
+
+<% content_for :extra_headers do %>
+  <%= render "govuk_publishing_components/components/machine_readable_metadata", { content_item: content_item.to_h, schema: :article } %>
```
@sairamya93 and I were just checking the metadata and noticed the meta `description` tag needs to be included here, similar to https://github.com/alphagov/frontend/pull/4509/files#diff-ee902e589d40510716bed23cb80d0a765570d0867944a3a9ea98cffcf34fa1afR10

leenagupte commented on spec/models/case_study_spec.rb:
```diff
@@ -0,0 +1,8 @@
+RSpec.describe CaseStudy do
+  it_behaves_like "it has updates", "case_study", "has-updates"
+  it_behaves_like "it has updates", "case_study", "only-has-first-published-at"
```
I see that you have updated the "only-has-first-published-at" example so that the `published_at` date is later than `first_published_at`, but doesn't have any updates, so the example is still incorrect.

If you update the example to have a change history, then it becomes the same test as for "has-updates". 

If you change `published_at` to equal `first_published_at`, then it becomes the same test as "doing-business-in-spain".

I think perhaps the "only-has-first-published-at" example and this line are not needed and should be removed.

leenagupte commented on spec/models/case_study_spec.rb:
```diff
@@ -0,0 +1,29 @@
+  it_behaves_like "it has updates", "case_study", "has-updates"
+  it_behaves_like "it has updates", "case_study", "only-has-first-published-at"
+  it_behaves_like "it has no updates", "case_study", "doing-business-in-spain"
+  it_behaves_like "it has no updates", "case_study", "no-public-updated-at"
```
The "no-public-updated-at" is bogus as there should always be a value for `public_updated_at` so I think this test and the associated example can be removed.

leenagupte commented on spec/models/case_study_spec.rb:
```diff
@@ -0,0 +1,29 @@
+RSpec.describe CaseStudy do
+  let(:content_item) do
```
To be consistent with other tests this should be called `content_store_response`.

leenagupte commented on spec/models/case_study_spec.rb:
```diff
@@ -0,0 +1,29 @@
+
+  describe "#contributors" do
+    it "returns the organisations ordered by emphasis" do
+      model_instance = described_class.new(content_item)
```
To be consistent with other tests this variable should be called `content_item` rather than `model_instance`

e.g. 
```ruby
content_item = described_class.new(content_store_response)
```

leenagupte commented on spec/models/case_study_spec.rb:
```diff
@@ -0,0 +1,29 @@
+  it_behaves_like "it can have emphasised organisations", "case_study", "doing-business-in-spain"
+
+  describe "#contributors" do
+    it "returns the organisations ordered by emphasis" do
```
This should be renamed to include "worldwide organisations". Perhaps something like:

`it "returns the organisations ordered by emphasis followed by worldwide organisations"`

leenagupte commented on spec/system/case_study_spec.rb:
```diff
@@ -0,0 +1,41 @@
+      expect(page).to have_css("h1", text: "Get Britain Building: Carlisle Park")
+      expect(page).to have_text("Nearly 400 homes are set to be built on the site of a former tar distillery thanks to Gleeson Homes and HCA investment.")
+
+      expect(page).to have_css(".gem-c-translation-nav")
```
I think this should be extracted into it's own context, rather than being tested by stealth or being shoe-horned into the withdrawn context. Perhaps something like `context "when the page has translations"`

leenagupte commented on spec/system/case_study_spec.rb:
```diff
@@ -0,0 +1,41 @@
+      expect(page).to have_css(".gem-c-translation-nav")
+    end
+
+    context "when visiting a Withdrawn Case Study page which is also translatable" do
```
As mentioned above, the "translatable" part of this context should be extracted into a separate context, along with the expectation.

leenagupte reviewed (commented):
Thanks for adding a test for the `contributors` method 🎉 

There are couple of other minor things in the tests that should be tidied up.

leenagupte reviewed (approved):
Thanks for making the requested changes. 👍 

I've checked a few case studies, and the pages seem to be rendering as expected and with the correct metadata.

A frontender should check this over as well before merging.

matthillco reviewed (approved):
I've looked at the visual changes and the frontend code: all looks good to me. Nice work.
